File: examples/student.cc
#include <string> 
using namespace std;

class Student {


  public:
    Student(string name, string student_id, string programme) 
      : name(move(name)), student_id(move(student_id)), programme(move(programme))
    {};

    string association() {
      return associations.at(name);
    }

    string programme;
    string student_id;
    string programme;

    std::unordered_map<string, string> associations = ...;

};

File: examples/student.rs
use std::collections::HashMap;

struct Student {
    name: String,
    student_id: String,
    programme: String,
}

impl Student {
    pub fn new(name: String, student_id: String, programme: String) -> Self {
        Self{name, student_id, programme}
    }

    pub fn association(&self) -> Option<&String> {
        match self.programme {
            "Applied Physics" => "Arago",
            _ => unimplemented!()
        }
    }
}
